src/MonitorProfiler/Environment/ProfilerEnvironment.cpp

```diff
@@ -0,0 +1,50 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#include "ProfilerEnvironment.h"
+
+using namespace std;
+
+ProfilerEnvironment::ProfilerEnvironment(ICorProfilerInfo11* pCorProfilerInfo) :
+    m_pCorProfilerInfo(pCorProfilerInfo)
+{
+}
+
+STDMETHODIMP ProfilerEnvironment::GetEnvironmentVariable(const tstring name, tstring& value)
```

Comment:
Maybe I missed these, but add comments regarding the env caching and why this is necessary.